src/apps/rack_levels/services.py

```diff
@@ -256,3 +312,102 @@ async def delete_single_rack_level(session: AsyncSession, rack_level_id: str):
 
     await session.commit()
     return result
+
+
+async def add_single_stock_to_rack_level(
+    session: AsyncSession,
+    rack_level_id: str,
+    stock_schema: StockRackLevelInputSchema,
+    user_id: str,
+) -> dict[str, str]:
+    from src.apps.stocks.services.stock_services import (
+        manage_resources_state_when_managing_stocks,
+    )
+    from src.apps.stocks.services.user_stock_services import create_user_stock_object
+
+    if not (
+        rack_level_object := await if_exists(RackLevel, "id", rack_level_id, session)
+    ):
+        raise DoesNotExist(RackLevel.__name__, "id", rack_level_id)
+
+    stock_id = stock_schema.id
+    if not (stock_object := await if_exists(Stock, "id", stock_id, session)):
+        raise DoesNotExist(Stock.__name__, "id", stock_id)
+
+    if stock_object.is_issued:
+        raise CannotMoveIssuedStockException
+
+    if stock_object.rack_level_slot:
+        if stock_object.rack_level_slot.rack_level_id == rack_level_slot_id:
```

Comment:
== rack_level_id*

Reply:
in this case i would keep it cuz it randomly pick place in the rack level, so if the stock is already in the requested rack level its a nonsense to let the stock be placed in the same rack level on the random place